File: src/main/java/pl/training/module7/duplication/before/LegacyArtifactPublisher.java
package pl.training.module7.duplication.before;

import java.util.Locale;
import java.util.Objects;

public final class LegacyArtifactPublisher {
    public String publishSnapshot(String artifactName, int buildNumber) {
        Objects.requireNonNull(artifactName, "artifactName");
        if (artifactName.isBlank()) {
            throw new IllegalArgumentException("artifactName must not be blank");
        }
        if (buildNumber <= 0) {
            throw new IllegalArgumentException(
                    "buildNumber must be greater than zero");
        }

        String normalizedName = artifactName.strip().toLowerCase(Locale.ROOT);
        return normalizedName + ":" + buildNumber + "-SNAPSHOT";
    }

    public String publishRelease(String artifactName, int buildNumber) {
        Objects.requireNonNull(artifactName, "artifactName");
        if (artifactName.isBlank()) {
            throw new IllegalArgumentException("artifactName must not be blank");
        }
        if (buildNumber <= 0) {
            throw new IllegalArgumentException(
                    "buildNumber must be greater than zero");
        }

        String normalizedName = artifactName.strip().toLowerCase(Locale.ROOT);
        return normalizedName + ":" + buildNumber;
    }
}

File: typescript/src/module7/duplication/before/LegacyArtifactPublisher.ts
import { IllegalArgumentError } from '../../../shared/errors.js';
import { requireNonNull } from '../../../shared/requireNonNull.js';

export class LegacyArtifactPublisher {
  publishSnapshot(artifactName: string, buildNumber: number): string {
    requireNonNull(artifactName, 'artifactName');
    if (artifactName.trim() === '') {
      throw new IllegalArgumentError('artifactName must not be blank');
    }
    if (buildNumber <= 0) {
      throw new IllegalArgumentError(
        'buildNumber must be greater than zero');
    }

    const normalizedName = artifactName.trim().toLowerCase();
    return normalizedName + ':' + buildNumber + '-SNAPSHOT';
  }

  publishRelease(artifactName: string, buildNumber: number): string {
    requireNonNull(artifactName, 'artifactName');
    if (artifactName.trim() === '') {
      throw new IllegalArgumentError('artifactName must not be blank');
    }
    if (buildNumber <= 0) {
      throw new IllegalArgumentError(
        'buildNumber must be greater than zero');
    }

    const normalizedName = artifactName.trim().toLowerCase();
    return normalizedName + ':' + buildNumber;
  }
}
